src/Files.App.Storage/Watchers/DeviceWatcher.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Microsoft.Management.Infrastructure;
+using Windows.Devices.Enumeration;
+using Windows.Devices.Portable;
+
+namespace Files.App.Storage.Watchers
+{
+	/// <inheritdoc cref="IDeviceWatcher"/>
+	public class DeviceWatcher : IDeviceWatcher
+	{
+		private Windows.Devices.Enumeration.DeviceWatcher? _winDeviceWatcher;
+
+		private ManagementEventWatcher? _mmiInsertWatcher, _mmiRemoveWatcher, _mmiModifyWatcher;
+
+		/// <inheritdoc/>
+		public bool CanBeStarted
+			=> _winDeviceWatcher?.Status is DeviceWatcherStatus.Created or DeviceWatcherStatus.Stopped or DeviceWatcherStatus.Aborted;
+
+		/// <inheritdoc/>
+		public event EventHandler<DeviceEventArgs>? ItemAdded;
```

Comment:
What do item refer to?

Reply:
The device, in this context, the drive.
Do you think it should be simply 'Added'? Other watchers have 'Item' prefixes so I'd like to rename all after this PR in another PR.

Reply:
It should be `DeviceAdded`

Reply:
Done.